src/hotspot/share/jfr/support/jfrIntrinsics.cpp

```diff
@@ -74,6 +74,7 @@ void* JfrIntrinsicSupport::write_checkpoint(JavaThread* jt) {
 
 void* JfrIntrinsicSupport::return_lease(JavaThread* jt) {
   DEBUG_ONLY(assert_precondition(jt);)
+  MACOS_AARCH64_ONLY(ThreadWXEnable __wx(WXWrite, jt));
```

Comment:
It seems like this could be moved down.  It doesn't seem to be needed for the Java --> native transition.  Is it needed for the JfrJavaEventWriter::flush() call?

Reply:
If it is only needed for the native --> Java transition below, why don't we do it lazily?  The interpreter and compilers already do this by calling check_special_condition_for_native_trans() only if a safepoint is detected.
Normally we would want to be in the WXExec state when executing in _thread_in_native.

Reply:
`WXWrite` is needed for 

```
JfrIntrinsicSupport::return_lease ->
ThreadStateTransition::transition_from_native ->
SafepointMechanism::process_if_requested_with_exit_check ->
SafepointMechanism::process_if_requested ->
JavaThread::check_possible_safepoint ->
assert_wx_state(WXWrite)
```

Reply:
> Normally we would want to be in the WXExec state when executing in _thread_in_native.
I agree. So we would need to aquire `WXWrite` twice just for `ThreadStateTransition::transition_from_java` and again for `ThreadStateTransition::transition_from_native`. I think its a bit unfortune that `WXWrite` is needed for the state transition..

Reply:
I don't think we need WXWrite for transition_from_java.  I don't know how useful AssertWXAtThreadSync is if it forces us to make unnecessary transitions.  It seems to go in the opposite direction from the more lazy approaches discussed in JDK-8307817.